src/bun.js/test/jest.zig

```diff
@@ -1737,3 +1772,187 @@ pub fn printGithubAnnotation(exception: *JSC.ZigException) void {
+                return .zero;
+            }
+            timeout_ms = @as(u32, @intCast(@max(timeout.coerce(i32, globalThis), 0)));
+        }
+        if (options.get(globalThis, "retry")) |retries| {
+            if (!retries.isNumber()) {
+                globalThis.throwPretty("{s} expects retry to be a number", .{signature});
+                return .zero;
+            }
+            // TODO: retry_count = @intCast(u32, @max(retries.coerce(i32, globalThis), 0));
+        }
+        if (options.get(globalThis, "repeats")) |repeats| {
+            if (!repeats.isNumber()) {
+                globalThis.throwPretty("{s} expects repeats to be a number", .{signature});
+                return .zero;
+            }
+            // TODO: repeat_count = @intCast(u32, @max(repeats.coerce(i32, globalThis), 0));
+        }
+    } else if (!options.isEmptyOrUndefinedOrNull()) {
+        globalThis.throwPretty("{s} expects options to be a number or object", .{signature});
+        return .zero;
+    }
+
+    const parent = DescribeScope.active.?;
+
+    if (JSC.getFunctionData(callee)) |data| {
+        const allocator = getAllocator(globalThis);
+        const each_data = bun.cast(*EachData, data);
+        JSC.setFunctionData(callee, null);
+        const array = each_data.*.strong.get() orelse return .zero;
+        defer {
+            each_data.*.strong.deinit();
+            allocator.destroy(each_data);
+        }
+
+        if (array.isUndefinedOrNull() or !array.jsType().isArray()) {
+            return .zero;
+        }
+
+        var iter = array.arrayIterator(globalThis);
+
+        while (iter.next()) |item| {
+            // TODO: node:util.format() the label
+            const label = if (description.isEmptyOrUndefinedOrNull())
+                ""
+            else
+                (description.toSlice(globalThis, allocator).cloneIfNeeded(allocator) catch unreachable).slice();
+
+            const func_params_length = function.getLength(globalThis);
+            const item_is_array = !item.isEmptyOrUndefinedOrNull() and item.jsType().isArray();
+            var arg_size: usize = 1;
+
+            if (item_is_array) {
+                arg_size = item.getLength(globalThis);
+            }
+
+            // add room for callback function
+            const has_callback_function: bool = (func_params_length > arg_size) and each_data.is_test;
+            if (has_callback_function) {
+                arg_size += 1;
+            }
+
+            var function_args = allocator.alloc(JSC.JSValue, arg_size) catch @panic("can't create function_args");
+            var idx: u32 = 0;
+
+            if (item_is_array) {
+                // Spread array as args
+                var item_iter = item.arrayIterator(globalThis);
+                while (item_iter.next()) |array_item| {
+                    if (array_item == .zero) {
+                        allocator.free(function_args);
```

Comment:
this leaks memory because existing function args will not be unprotected 

but lets handle that a little later, this is very much an edgecase as-is